src/extension/module.js

```diff
@@ -1432,14 +1432,52 @@
     }
   }
 
+  /**
+   * Registers a plugin for re-evaluation if it should be shown or hidden,
+   * and if its button should be enabled or disabled.
+   * @private
+   * @param {Sidekick} sk The sidekick
+   * @param {Plugin} plugin The plugin configuration
+   * @param {HTMLElement} $plugin The plugin
+   * @returns {HTMLElement} The plugin or {@code null}
+   */
+  function registerPlugin(sk, plugin, $plugin) {
+    // re-evaluate plugin when status fetched
+    sk.addEventListener('statusfetched', () => {
+      if (typeof plugin.condition === 'function') {
+        if ($plugin && !plugin.condition(sk)) {
+          // plugin exists but condition now false
+          sk.remove(plugin.id);
+        } else if (!$plugin && plugin.condition(sk)) {
+          // plugin doesn't exist but condition now true
+          sk.add(plugin);
+        }
+      }
+      const isEnabled = plugin.button && plugin.button.isEnabled;
+      if (typeof isEnabled === 'function' || typeof isEnabled === 'boolean') {
+        const $button = $plugin && $plugin.querySelector(':scope button');
+        if ($button) {
+          if (typeof isEnabled === 'function' && isEnabled(sk)) {
+            // button enabled
+            $plugin.querySelector(':scope button').removeAttribute('disabled');
+          } else {
+            // button disabled
+            $plugin.querySelector(':scope button').setAttribute('disabled', '');
+          }
+        }
+      }
+    });
+    return $plugin || null;
+  }
+
   /**
    * Checks existing plugins based on the status of the current resource.
    * @private
    * @param {Sidekick} sk The sidekick
    */
   function checkPlugins(sk) {
     window.setTimeout(() => {
-      if (sk.pluginContainer.querySelectorAll(':scope > div > *').length === 0) {
+      if (sk.pluginContainer.querySelectorAll(':scope div.plugin').length === 0) {
```

Comment:
why `querySelectorAll` ? in general: if you have many matches, the `querySelectorAll` would be slower than just finding the first one.

```suggestion
      if (!sk.pluginContainer.querySelector(':scope div.plugin')) {
```